Fix expected, actual links for variant-based imported wpt tests

philn opened the pull request:
#### 3be2c0d6c2ab7ebbf322a81a391cd2627c687d77
<pre>
Fix expected, actual links for variant-based imported wpt tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=238832">https://bugs.webkit.org/show_bug.cgi?id=238832</a>
&lt;rdar://problem/91313891&gt;

Reviewed by Jonathan Bedard.

Ensure that the workaround involving &quot;len(fs.splitext(output_basename)[1]) - 1 &gt; 5&quot;
does not affect imported templated wpt tests that may match that condition, such as
&quot;.../pbkdf2.https.any.worker.html.&quot;

A unittest for SVG results was added as well, since the initial patch was reverted due to this case.

Based on rolled out patch authored by J Pascoe  &lt;j_pascoe@apple.com&gt;.

* LayoutTests/fast/harness/results.html:
* Tools/Scripts/webkitpy/layout_tests/controllers/test_result_writer.py:
(TestResultWriter.output_filename):
* Tools/Scripts/webkitpy/layout_tests/controllers/test_result_writer_unittest.py:
(TestResultWriterTest.test_output_filename):
(TestResultWriterTest):
(TestResultWriterTest.test_output_filename_worker_variant):
(TestResultWriterTest.test_output_filename_variant):
(TestResultWriterTest.test_output_svg_filename):

Canonical link: <a href="https://commits.webkit.org/256703@main">https://commits.webkit.org/256703@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c079afebcbb4dca64715e2267397dcf53c46cfe6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/96587 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/5843 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/29664 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106113 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/166448 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/100571 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6037 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/34581 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88960 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/102824 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/102262 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/4498 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83190 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/31471 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/100178 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/86341 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/88221 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/74378 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/40312 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/19710 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37975 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/21115 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4651 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/4244 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/43659 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1057 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/40391 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Tools/Scripts/webkitpy/layout_tests/controllers/test_result_writer.py:
```diff
@@ -100,18 +100,22 @@ def output_filename(self, modifier):
 
         ext_parts = fs.splitext(self._test_name)
         output_basename = ext_parts[0]
+        extension = ext_parts[1]
```
What is the guarantee that ext_parts has more than 1 entry?

philn replied:
It's an API guarantee from Python, see https://docs.python.org/3/library/os.path.html?highlight=os%20path%20splitext#os.path.splitext

youennf commented on Tools/Scripts/webkitpy/layout_tests/controllers/test_result_writer.py:
```diff
@@ -100,18 +100,22 @@ def output_filename(self, modifier):
+        for char in ('?', '#'):
+            index = extension.find(char)
+            if index != -1:
+                extension = extension[:index]
```
Where is `extension` used?

philn replied:
Hum, good question, I wonder if it was intended to be used instead of `ext_parts[1]` below. I'll take a look!

JonWBedard replied:
That's can't be the case, though, because otherwise we'd end up clobbering the same file name for different test variants. It looks like the intent of this code is to change something like `foo/bar.html?variant=1` to have a result file of something like `foo/bar_variant=1-expected.txt`

philn replied:
Yes, I updated the patch and added a comment about this, hopefully it looks better now.

JonWBedard reviewed (approved):
Verified this locally, it does work! :)

philn committed:
Fix expected, actual links for variant-based imported wpt tests
https://bugs.webkit.org/show_bug.cgi?id=238832
<rdar://problem/91313891>

Reviewed by Jonathan Bedard.

Ensure that the workaround involving "len(fs.splitext(output_basename)[1]) - 1 > 5"
does not affect imported templated wpt tests that may match that condition, such as
".../pbkdf2.https.any.worker.html."

A unittest for SVG results was added as well, since the initial patch was reverted due to this case.

Based on rolled out patch authored by J Pascoe  <j_pascoe@apple.com>.

* LayoutTests/fast/harness/results.html:
* Tools/Scripts/webkitpy/layout_tests/controllers/test_result_writer.py:
(TestResultWriter.output_filename):
* Tools/Scripts/webkitpy/layout_tests/controllers/test_result_writer_unittest.py:
(TestResultWriterTest.test_output_filename):
(TestResultWriterTest):
(TestResultWriterTest.test_output_filename_worker_variant):
(TestResultWriterTest.test_output_filename_variant):
(TestResultWriterTest.test_output_svg_filename):

Canonical link: https://commits.webkit.org/256703@main
